compiler/cpp/src/parser/parser.cpp

```diff
@@ -27,14 +27,22 @@ std::unique_ptr<NumberExprAST> parseNumber() {
     return std::make_unique<NumberExprAST>(value);
 }
 
+std::unique_ptr<BooleanExprAST> parseBoolean() {
+    bool value = CurTok.getValue() == "true";
+    getNextToken(); // Consume boolean literal
+    return std::make_unique<BooleanExprAST>(value);
+}
+
 std::unique_ptr<StringExprAST> parseString() {
     std::string value = CurTok.getValue();
     getNextToken(); // Consume string literal
     return std::make_unique<StringExprAST>(value);
 }
 
 std::unique_ptr<ValueExprAST> parseValue() {
-    if (CurTok.getType() == TOK_NUMBER) return parseNumber(); // Consume number
+    int type = CurTok.getType();
+    if (type == TOK_NUMBER) return parseNumber(); // Consume number
+    if (type == TOK_TRUE || type == TOK_FALSE) return parseBoolean(); // Consume boolean
```

Comment:
Is this reflected in our grammar already? I only spotted: `VALUE                 --> STRING | NUMBER`

Reply:
Oh. Good point! I forgot about the grammar.
Will update the grammar in a min ...